utils/treesitter/grammar.js

```diff
@@ -0,0 +1,407 @@
+/*
+ * Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+ * Exceptions. See /LICENSE for license information.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+function repeat_sep1(thing, sep) {
+  return seq(thing, repeat(seq(sep, thing)));
+}
+
+function repeat_sep(thing, sep) {
```

Comment:
Looks like in all uses, sep is `,` and this is followed by an optional `,`. Perhaps a `comma_sep` function would be better?